app/models/channel/filter/database.rb

```diff
@@ -18,12 +18,12 @@ def self.run(_channel, mail)
           match_rule = meta['value']
           min_one_rule_exists = true
           if meta[:operator] == 'contains not'
-            if value.present? && match(value, match_rule, false)
+            if value.present? && Channel::Filter::Match::EmailRegex.match(value: value, match_rule: match_rule, check_mode: false)
               all_matches_ok = false
               Rails.logger.info "  matching #{key.downcase}:'#{value}' on #{match_rule}, but shoud not"
             end
           elsif meta[:operator] == 'contains'
-            if value.blank? || !match(value, match_rule, true)
+            if value.blank? || !Channel::Filter::Match::EmailRegex.match(value: value, match_rule: match_rule, check_mode: true)
```

Comment:
Looks like the change of attributes has introduced a behavior change. The third parameter `true` was previously the now removed `_should_match` but is now `check_mode`.